src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13846,5 +13846,144 @@ int32[] arg2
+                dWithParams = (params int[] xs) => xs.Length;
+                dWithParams = (int[] xs) => xs.Length; // 2
+                dWithParams = MethodNoParams;
+                dWithParams = MethodWithParams;
+                delegate int DelegateNoParams(int[] xs);
+                delegate int DelegateWithParams(params int[] xs);
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (5,27): error CS9503: Parameter 1 has params modifier in lambda and not in target delegate type.
+                // dNoParams = (params int[] xs) => xs.Length; // 1
+                Diagnostic(ErrorCode.ERR_ParamsArrayInLambdaOnly, "xs").WithArguments("1").WithLocation(5, 27),
+                // (10,22): error CS9504: Parameter 1 has params modifier in target delegate type and not in lambda.
+                // dWithParams = (int[] xs) => xs.Length; // 2
+                Diagnostic(ErrorCode.ERR_ParamsArrayInDelegateOnly, "xs").WithArguments("1").WithLocation(10, 22));
+        }
+
+        [Fact]
+        public void ParamsArray_CommonType()
+        {
+            var source = """
+                int MethodNoParams(int[] xs) => xs.Length;
+                int MethodWithParams(params int[] xs) => xs.Length;
+                var inferredNoParams = MethodNoParams;
+                var inferredWithParams = MethodWithParams;
+                var lambdaNoParams = (params int[] xs) => xs.Length;
+                var lambdaWithParams = (int[] xs) => xs.Length;
+                var a1 = new[] { MethodNoParams, MethodNoParams };
+                var a2 = new[] { inferredNoParams, inferredWithParams }; // 1
+                var a3 = new[] { lambdaNoParams, lambdaWithParams }; // 2
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,10): error CS0826: No best type found for implicitly-typed array
+                // var a2 = new[] { inferredNoParams, inferredWithParams }; // 1
+                Diagnostic(ErrorCode.ERR_ImplicitlyTypedArrayNoBestType, "new[] { inferredNoParams, inferredWithParams }").WithLocation(8, 10),
+                // (9,10): error CS0826: No best type found for implicitly-typed array
+                // var a3 = new[] { lambdaNoParams, lambdaWithParams }; // 2
+                Diagnostic(ErrorCode.ERR_ImplicitlyTypedArrayNoBestType, "new[] { lambdaNoParams, lambdaWithParams }").WithLocation(9, 10));
+        }
+
+        [Fact]
+        public void ParamsArray_CommonType_Methods()
+        {
+            var source = """
+                int MethodNoParams(int[] xs) => xs.Length;
+                int MethodWithParams(params int[] xs) => xs.Length;
+                var a = new[] { MethodNoParams, MethodNoParams };
+                System.Console.WriteLine(a.GetType());
+                """;
+            CompileAndVerify(source, expectedOutput: "System.Func`2[System.Int32[],System.Int32][]");
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Please test explicit delegate types if not covered already. Presumably, the first example below should be allowed since it is currently valid.
```csharp
D1 d1 = (int[] args) => { }; // ok
D2 d2 = (params int[] args) => { }; // error
delegate void D1(params int[] args);
delegate void D2(int[] args);
```

Reply:
They are covered few lines above:

https://github.com/dotnet/roslyn/pull/64961/files/7747e2bb1d6547788b83a588cc0c680bd74597e8#diff-5be902de0bb9ce3b40465571a994902a42d338ba661eb92226b84626ceb8e1ffR13932-R13943

The first example is currently an error, though, good catch, I will fix that.